torch/_dynamo/variables/builder.py

```diff
@@ -440,6 +440,31 @@ def index_source(key):
                 guards=make_guards(GuardBuilder.FUNCTION_MATCH),
             )
         # NB: These can't be put in type_dispatch, they have to run later
+        elif value in {
+            torch.distributed.distributed_c10d.all_gather_into_tensor,
+            torch.distributed.distributed_c10d.reduce_scatter_tensor,
+        }:
+            from torch.distributed._functional_collectives import (
+                all_gather_tensor_inplace,
+                reduce_scatter_tensor_inplace,
+            )
+            from torch.distributed.distributed_c10d import (
+                all_gather_into_tensor as legacy_allgather,
+                reduce_scatter_tensor as legacy_reducescatter,
+            )
+
+            # TODO assert that async_op is false, and, possibly some other limitations
+            # current remapping is very brittle, in regards to handling args at call time
+            # and seriously.. can't import properly, and also getting line-too-long linter
+            traceable_collective_remaps = {
+                legacy_allgather: all_gather_tensor_inplace,
+                legacy_reducescatter: reduce_scatter_tensor_inplace,
+            }
+            return UserFunctionVariable(
```

Comment:
Does this inline through it? Can you dump the fx graph? I am not sure this is what we want...

Reply:
yes, this inlines through it.

It's good bc i wrote it as a user func- it calls first `copy_` and then `all_gather_blah` both of which are traceable.  

I looked at the fx graph from running my unit test with `TORCH_LOGS=graph_code` or whatever the right name is.  IT showed 3 ops - all_gather, wait, copy_.